modules/op-csi-secrets.adoc

```diff
@@ -0,0 +1,167 @@
+- kind: ServiceAccount
+  name: pipeline-sa
+  namespace: tekton-vault
+----
+
+. Create a `SecretProviderClass` CR that defines the secrets that your pipeline or task consumes and the mount path and filename (key) for each of these secrets.
++
+.Example definition of the `vault-secret` `SecretProviderClass` CR
+[source,yaml]
+----
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: vault-secret
+  namespace: tekton-vault
+spec:
+  provider: vault
+  parameters:
+    vaultAddress: "http://vault.vault:8200"
+    vaultSkipTLSVerify: "true"
+    roleName: "my-role"
+    objects: |
+      - objectName: "demo-secret"
+        secretPath: "secret/data/my-secret"
+        secretKey: "username"
+      - objectName: "demo-secret-pass"
+        secretPath: "secret/data/my-secret"
+        secretKey: "password"
+----
++
+[NOTE]
+====
+The `vaultAddress` parameter used in this example uses `http` and `vaultSkipTLSVerify`: `"true"` only for demonstration purposes. For production use, configure Vault with a secure `https` endpoint and do not skip TLS verification.
+====
+
+. In the definition of the task that consumes the secrets, define and mount a volume that uses the CSI secrets store driver and specifies the name of the `SecretProviderClass` CR. To minimise security exposure of secrets, mount them in the definition of a task and not of an entire pipeline.
++
+.Example definition of a task that consumes secrets
+[source,yaml]
+----
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: secret-task
+  namespace: tekton-vault
+spec:
+  steps:
+    - name: use-secret
+      image: registry.access.redhat.com/ubi8/ubi-minimal:latest
+      script: |
+        #!/bin/sh
+        echo "Reading secrets from mounted volume..."
+        echo "Username: $(cat /mnt/secrets-store/demo-secret)"
+        echo "Password: $(cat /mnt/secrets-store/demo-secret-pass)"
+      volumeMounts:
+        - name: secrets-store-inline
+          mountPath: "/mnt/secrets-store"
+          readOnly: true
+  volumes:
+    - name: secrets-store-inline
+      csi:
+        driver: secrets-store.csi.k8s.io
+        readOnly: true
+        volumeAttributes:
+          secretProviderClass: "vault-secret"
+----
++
+[NOTE]
+====
+The mount path that you specify in the pipeline or task definition overrides the path that you specify in the `SecretProviderClass` CR.
+====
+
+. In the `PipelineRun` or `TaskRun` CR, assign the service account that is authenticated with the secret store to the task:
+
+** If you create a `PipelineRun` CR, use the `taskRunSpecs` section to assign the service account to the particular task. Do not assign the service account to the entire pipeline.
++
+.Example `PipelineRun` CR
+[source,yaml]
+----
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: vault-secret-pipeline
+  namespace: tekton-vault
+spec:
+  tasks:
+    - name: read-secret
+      taskRef:
+        name: secret-task
+      serviceAccountName: pipeline-sa
```

Comment:
```suggestion
apiVersion: tekton.dev/v1
kind: Pipeline
metadata:
  name: vault-secret-pipeline
  namespace: tekton-vault
spec:
  tasks:
    - name: read-secret
      taskRef:
        name: secret-task
      serviceAccountName: pipeline-sa
---
apiVersion: tekton.dev/v1
kind: PipelineRun
metadata:
  name: vault-secret-pipeline-run
  namespace: tekton-vault
spec:
  pipelineRef:
    name: vault-secret-pipeline
  taskRunSpecs:
    - pipelineTaskName: secret-task
      serviceAccountName: pipeline-sa
```

Reply:
Both pipeline and pipelinerun need to be there in this case

Reply:
Fixed.